motion/thruster_interface_auv_cpp/include/thruster_interface_auv_cpp/thruster_interface_auv_driver.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef THRUSTER_INTERFACE_AUV_DRIVER_HPP
+#define THRUSTER_INTERFACE_AUV_DRIVER_HPP
+
+#include <vector>
+#include <string>
+#include <map>
+#include <iostream>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <linux/i2c-dev.h>
+#include <cmath>
+
+class ThrusterInterfaceAUVDriver {
+public:
+
+    //default contructor called when instantiating the object in .hpp file
+    ThrusterInterfaceAUVDriver();
+    ~ThrusterInterfaceAUVDriver();
+    
+    //contructor actually used
+    ThrusterInterfaceAUVDriver(
+        int I2C_BUS,
+        int PICO_I2C_ADDRESS,
+        double SYSTEM_OPERATIONAL_VOLTAGE,
+        const std::vector<int>& THRUSTER_MAPPING,
+        const std::vector<int>& THRUSTER_DIRECTION,
+        const std::vector<int>& THRUSTER_PWM_OFFSET,
+        const std::vector<int>& PWM_MIN,
+        const std::vector<int>& PWM_MAX,
+        const std::map<int, std::map<std::string, std::vector<double>>>& COEFFS
+    );
```

Comment:
Another side note, when constructing the `ThrusterParameters` objects, you can do additional checks to ensure that values are within legal range!

Reply:
why passing a struct as std::vector<struct> ? when i have to pass only one struct and not many? to be able to pass them by reference?

Reply:
Conceptually each element of the current THRUSTER_MAPPING/DIRECTION etc etc corresponds to one specific thruster. It would be nicer to let the constructor look something like
```C++
    ThrusterInterfaceAUVDriver(
        int i2c_bus,
        int pico_i2c_addresss,
        const std::vector<ThrusterParameters>& thruster_parameters,
        const std::map<int, std::map<std::string, std::vector<double>>>& thrust_mapping_coefficients
    );
```

If you pass a bunch of separate std::vectors<> to the constructor, it's really easy to mess up the order and stuff. Doing it this way lets you construct the `ThrusterParameters` in the same place you read the parameters from file. Here, you can also perform checks on each thruster's parameters, to ensure that f.ex. the PWM values are inside legal ranges.


The TLDR: When you end up with many (5+ as a rule of thumb, but it's case by case really) input parameters, and/or these inputs are of the same type, you probably want to place them in a struct, and pass that instead.